File: src/frontend/patrimonio/lib/app/data/model/patrim_grupo_bem_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';


class PatrimGrupoBemModel {
	int? id;
	String? codigo;
	String? nome;
	String? descricao;
	String? contaAtivoImobilizado;
	String? contaDepreciacaoAcumulada;
	String? contaDespesaDepreciacao;
	int? codigoHistorico;

	PatrimGrupoBemModel({
		this.id,
		this.codigo,
		this.nome,
		this.descricao,
		this.contaAtivoImobilizado,
		this.contaDepreciacaoAcumulada,
		this.contaDespesaDepreciacao,
		this.codigoHistorico,
	});

	static List<String> dbColumns = <String>[
		'id',
		'codigo',
		'nome',
		'descricao',
		'conta_ativo_imobilizado',
		'conta_depreciacao_acumulada',
		'conta_despesa_depreciacao',
		'codigo_historico',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Codigo',
		'Nome',
		'Descricao',
		'Conta Ativo Imobilizado',
		'Conta Depreciacao Acumulada',
		'Conta Despesa Depreciacao',
		'Codigo Historico',
	];

	PatrimGrupoBemModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		codigo = jsonData['codigo'];
		nome = jsonData['nome'];
		descricao = jsonData['descricao'];
		contaAtivoImobilizado = jsonData['contaAtivoImobilizado'];
		contaDepreciacaoAcumulada = jsonData['contaDepreciacaoAcumulada'];
		contaDespesaDepreciacao = jsonData['contaDespesaDepreciacao'];
		codigoHistorico = jsonData['codigoHistorico'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['codigo'] = codigo;
		jsonData['nome'] = nome;
		jsonData['descricao'] = descricao;
		jsonData['contaAtivoImobilizado'] = contaAtivoImobilizado;
		jsonData['contaDepreciacaoAcumulada'] = contaDepreciacaoAcumulada;
		jsonData['contaDespesaDepreciacao'] = contaDespesaDepreciacao;
		jsonData['codigoHistorico'] = codigoHistorico;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		codigo = plutoRow.cells['codigo']?.value;
		nome = plutoRow.cells['nome']?.value;
		descricao = plutoRow.cells['descricao']?.value;
		contaAtivoImobilizado = plutoRow.cells['contaAtivoImobilizado']?.value;
		contaDepreciacaoAcumulada = plutoRow.cells['contaDepreciacaoAcumulada']?.value;
		contaDespesaDepreciacao = plutoRow.cells['contaDespesaDepreciacao']?.value;
		codigoHistorico = plutoRow.cells['codigoHistorico']?.value;
	}	

	PatrimGrupoBemModel clone() {
		return PatrimGrupoBemModel(
			id: id,
			codigo: codigo,
			nome: nome,
			descricao: descricao,
			contaAtivoImobilizado: contaAtivoImobilizado,
			contaDepreciacaoAcumulada: contaDepreciacaoAcumulada,
			contaDespesaDepreciacao: contaDespesaDepreciacao,
			codigoHistorico: codigoHistorico,
		);			
	}

	
}

File: backend/patrimonio/src/model/patrim_grupo_bem_model.py
from src import db


class PatrimGrupoBemModel(db.Model):
    __tablename__ = 'patrim_grupo_bem'

    id = db.Column(db.Integer, primary_key=True)
    codigo = db.Column(db.String(3))
    nome = db.Column(db.String(50))
    descricao = db.Column(db.Text)
    conta_ativo_imobilizado = db.Column(db.String(30))
    conta_depreciacao_acumulada = db.Column(db.String(30))
    conta_despesa_depreciacao = db.Column(db.String(30))
    codigo_historico = db.Column(db.Integer)


    def mapping(self, data):
        self.id = data.get('id')
        self.codigo = data.get('codigo')
        self.nome = data.get('nome')
        self.descricao = data.get('descricao')
        self.conta_ativo_imobilizado = data.get('contaAtivoImobilizado')
        self.conta_depreciacao_acumulada = data.get('contaDepreciacaoAcumulada')
        self.conta_despesa_depreciacao = data.get('contaDespesaDepreciacao')
        self.codigo_historico = data.get('codigoHistorico')

    def serialize(self):
        return {
            'id': self.id,
            'codigo': self.codigo,
            'nome': self.nome,
            'descricao': self.descricao,
            'contaAtivoImobilizado': self.conta_ativo_imobilizado,
            'contaDepreciacaoAcumulada': self.conta_depreciacao_acumulada,
            'contaDespesaDepreciacao': self.conta_despesa_depreciacao,
            'codigoHistorico': self.codigo_historico,
        }